implementations/rust/ockam/ockam/src/profile.rs

```diff
@@ -191,18 +191,98 @@ impl Profile {
     pub const CURRENT_CHANGE_VERSION: u8 = 1;
 }
 
-impl Profile {
+pub trait ProfileIdentity {
     /// Return unique [`Profile`] identifier, which is equal to sha256 of the root public key
-    pub fn identifier(&self) -> &ProfileIdentifier {
+    fn identifier(&self) -> &ProfileIdentifier;
+}
+
+pub trait ProfileChanges {
+    /// Return change history chain
+    fn change_events(&self) -> &[ProfileChangeEvent];
+    /// Add a change event.
+    fn update_no_verification(
```

Comment:
I feel like this should be renamed, the current name here is unclear. `add_change_event`? `add_change_no_verify`? How critical is the 'no verification' aspect? I don't see an alternative which updates and also verifies..